### ISSUE TITLE
Plot cellular density and gene expression along axis

### ISSUE THREAD

**Issue by jwrth** [SpatialPathology/InSituPy#174]
Package Mellon uses gaussian kernel density to assess the phenotypic density along pseudotime:
![Image](https://github.com/user-attachments/assets/c97272a7-8891-4055-bb59-8f9629f28c74)

It can be used to plot cell types along axis and assess their density:
![Image](https://github.com/user-attachments/assets/4babae3a-4c28-40f1-b01e-8728d15d270d)

But also to evaluate gene expression:
![Image](https://github.com/user-attachments/assets/e2782bdb-2987-4a3d-8244-10d6fa830e1a)

We could either use the mellon estimator:
```Python
%%time
adata = xd.cells.matrix

model = mellon.DensityEstimator()
log_density = model.fit_predict(adata.obsm["spatial"])

predictor = model.predict

adata.obs["mellon_log_density"] = log_density
adata.obs["mellon_log_density_clipped"] = np.clip(
    log_density, *np.quantile(log_density, [0.05, 1])
)
```

OR we just use the standard kernel density. This one is maybe safer since we know what actually happens. Mellon might be optimized for high-dimensional data.

```Python
import numpy as np
import matplotlib.pyplot as plt
from sklearn.neighbors import KernelDensity

# Generate 2-dimensional bimodal data
np.random.seed(0)
X1 = np.random.normal(loc=[-2, -2], scale=0.5, size=(100, 2))
X2 = np.random.normal(loc=[2, 2], scale=0.5, size=(100, 2))
X = np.vstack([X1, X2])

# Define different bandwidths to illustrate
bandwidths = [0.1, 0.5, 1.0]

# Plot the data and the estimated density for different bandwidths
plt.figure(figsize=(15, 5))

for i, bandwidth in enumerate(bandwidths):
    kde = KernelDensity(kernel='gaussian', bandwidth=bandwidth).fit(X)
    X_plot = np.linspace(-5, 5, 100)
    Y_plot = np.linspace(-5, 5, 100)
    X_plot, Y_plot = np.meshgrid(X_plot, Y_plot)
    xy_plot = np.vstack([X_plot.ravel(), Y_plot.ravel()]).T
    log_dens = kde.score_samples(xy_plot).reshape(X_plot.shape)

    plt.subplot(1, len(bandwidths), i + 1)
    plt.contourf(X_plot, Y_plot, np.exp(log_dens), alpha=0.5)
    plt.scatter(X[:, 0], X[:, 1], s=10, color='k')
    plt.title(f'Bandwidth = {bandwidth}')
    plt.xlabel('X-axis')
    plt.ylabel('Y-axis')

plt.tight_layout()
plt.show()

```
![Image](https://github.com/user-attachments/assets/31d2b5b1-9cd8-4c9e-9b56-bb7e642d54bb)
